client/python/cryoet_data_portal/src/cryoet_data_portal/_codegen.py

```diff
@@ -292,6 +292,14 @@ def _parse_model_field(
     return None
 
 
+def _get_remote_field_name_by_type(
+    model_cls: GraphQLObjectType, lookup_type: GraphQLObjectType,
+) -> str | None:
```

Comment:
```suggestion
) -> Optional[str]:
```

Or bump the min version of Python from 3.7 to 3.10, which seems a tad aggressive.